packages/dx-react-grid/src/components/table-layout/virtual-row-layout.test.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import { shallow, mount } from 'enzyme';
+import { setupConsole } from '@devexpress/dx-testing';
+import { VirtualRowLayout } from './virtual-row-layout';
+
+const defaultProps = {
+  row: { key: 1, rowId: 1, height: 20 },
+  cells: [
+    { column: { key: 'a', column: { name: 'a' } }, colSpan: 1 },
+    { column: { key: 'b', column: { name: 'b' } }, colSpan: 1 },
+    { column: { key: 'c', column: { name: 'c' } }, colSpan: 1 },
+    { column: { key: 'd', column: { name: 'd' } }, colSpan: 1 },
+  ],
+  rowComponent: () => null,
+  cellComponent: () => null,
+};
+
+describe('RowLayout', () => {
+  let resetConsole;
+  beforeEach(() => {
+    resetConsole = setupConsole();
+  });
+
+  afterEach(() => {
+    resetConsole();
+    jest.resetAllMocks();
+  });
+
+  it('should render the "rowComponent" with correct properties', () => {
+    const tree = shallow((
+      <VirtualRowLayout
+        {...defaultProps}
+      />
+    ));
+
+    expect(tree.find(defaultProps.rowComponent).at(0).props())
+      .toMatchObject({
+        tableRow: defaultProps.row,
+        style: { height: '20px' },
+      });
+  });
+
+  it('should render the "cellComponent" for each column', () => {
+    const tree = shallow((
+      <VirtualRowLayout
+        {...defaultProps}
+      />
+    ));
+
+    tree.find(defaultProps.rowComponent).at(0).children().forEach((cell, index) => {
```

Comment:
We need to add `expect.hasAssertions();` to this test to ensure this assertion will be checked

Reply:
May be it would be simpler to check length of the array before checking each array item?